### ISSUE TITLE
Sharing scripts between projects

### ISSUE THREAD

**Issue by djanosik** [nightroman/Invoke-Build#35]
I have a simple build script (https://gitlab.com/snippets/25985) and I will use it in many projects without any changes. I would like to share it between projects. One way is to download the build script from somewhere and it would be nice if Invoke-Build could do this.

```powershell
InvokeBuild -Url "https://gitlab.com/snippets/25985/raw"
```

It will download and execute the build script. But of course, I can also do it that way.

```
Invoke-WebRequest "https://gitlab.com/snippets/25985/raw" -OutFile ".build.ps1"
Invoke-Build
```

**Comment by nightroman** [comment]
[PowerTasks]: https://github.com/shaynevanasperen/PowerTasks

This scenario is possible but not typical or unique. Build scripts are supposed to live together with their projects or be delivered by other tools. Delivering a whole script is not the only possible way. See [PowerTasks] which is based on Invoke-Build and uses nuget in order to get required common tasks from packages.

I think it is better if `Invoke-Build` is focused on building existing scripts and used as a pure build engine in more complex scenarios when needed. In your case it is just one more command, `Invoke-WebRequest`.

In other words, I am going to reject this feature request. The feature is not related to building and it covers just one extra case (useful but too simple) of several possible. Also, this feature would consume parameters potentially needed for something build related in the future (Url, Credentials, etc.).

**Comment by djanosik** [comment]
Ok, thank you!
